google/reader.go

```diff
@@ -0,0 +1,68 @@
+package google
+
+import (
+	"context"
+	"strings"
+
+	"google.golang.org/api/compute/v1"
+	"google.golang.org/api/option"
+)
+
+type GCPReader struct {
+	compute *compute.Service
+	project string
+	region  string
+	//TODO: add filters
+}
+
+// NewGCPReader returns a GCPReader with a catalog of services
+// ready to be used
+func NewGCPReader(ctx context.Context, project, region, credentials string) (*GCPReader, error) {
+	comp, err := compute.NewService(ctx, option.WithCredentialsFile(credentials))
+	if err != nil {
+		return nil, err
+	}
+	return &GCPReader{
+		compute: comp,
+		project: project,
+		region:  region,
+	}, nil
+}
+
+func (r *GCPReader) getZones() ([]string, error) {
+	rs := compute.NewRegionsService(r.compute)
+	region, err := rs.Get(r.project, r.region).Do()
+	if err != nil {
+		return nil, err
+	}
+	// zones are URL format. Need to split them
```

Comment:
Can you put an example? I see for the code what it is (it's the last element of the URL that we want) but just to have itmore clear.